### ISSUE TITLE
libssl1.1 wrong fix?

### ISSUE THREAD

**Issue by partizanes** [aquasecurity/trivy#4151]
![image](https://user-images.githubusercontent.com/48417/235485382-411abb8e-c603-4893-93c1-39b0be403c7e.png)

Hello, anyone can me explain why for libssl1.1 proposed fix as a package 3.0.2 version and how can fix that?

I have 
libssl1.1/now 1.1.1t amd64 [installed,local]
libssl3/jammy-updates,jammy-security,now 3.0.2-0ubuntu1.9 amd64 [installed,automatic]
openssl/jammy-updates,jammy-security,now 3.0.2-0ubuntu1.9 amd64 [installed]

Thank you!

**Comment by DmitriyLewen** [comment]
Hello @partizanes 
Thanks for your interest to Trivy!

For OS packages(`dpkg`, `apk`, `rpm`) we [use advisories of vendor OS](https://aquasecurity.github.io/trivy/v0.41/docs/scanner/vulnerability/os/#data-source-selection).
Source name of [libssl1.1](https://packages.ubuntu.com/en/focal/libssl1.1) is `openssl`:
 
![image](https://user-images.githubusercontent.com/91113035/235583194-e147e271-9257-46a3-a904-6260b312f9b6.png)
That is why fixed version is 3.0.2.

> how can fix that?

You need to update/remove this package.

Regards, Dmitriy

**Comment by partizanes** [comment]
@DmitriyLewen 
At first I try to install all last update from official repo.When it didn't work i manualy build  deb package sslib version 1.1.1t and install it. Also i have 3.0.2-0ubuntu1.9 amd64. The version > fixed version on screen.. It turns out that I have all the latest versions and still get an error. I can't remove libssl1.1 it because it has dependencies in the OS (ubuntu-minimal, python3 and etc ). Maybe I do not understand something and I will be grateful to you if you tell me.

**Comment by DmitriyLewen** [comment]
you have 2 package:
1) libssl1.1. `libssl1.1` is part of `openssl` source package - https://packages.ubuntu.com/en/source/focal/openssl. This source package contains vulnerabilities that was fixed in 3.0.2 version. So your `libssl1.1` library contains these vulnerabilities.
Also i couldn't find `1.1.1t` version. Looks like versions `1.1.1x` are only used for `focal` ubuntu. If i understand correctly you need to use `libssl3` for `jammy` ubuntu - https://packages.ubuntu.com/en/source/jammy/openssl
2) openssl. This library has actual version and doesn't have vulnerabilities.

**Comment by partizanes** [comment]
> 

Thank you for answer!

But i already have libssl3(.0.2-0ubuntu1.9 amd64) and can`t remove libssl1.1 because have dependency in ubuntu OS 22.04  (ubuntu-minimal, python3 and etc )

apt list --installed | grep ssl
```
libssl1.1/now 1.1.1f-1ubuntu2 amd64 [installed,local]
libssl3/jammy-updates,jammy-security,now 3.0.2-0ubuntu1.9 amd64 [installed,automatic]
openssl/jammy-updates,jammy-security,now 3.0.2-0ubuntu1.9 amd64 [installed]
```
At the beginning, 1.1.1f-1ubuntu2 was installed on the system and to update it, I downloaded the openssl-1.1.1t.tar.gz sources, compiled them and selected the files needed for libssl1.1, after which I built the libssl1.1_1.1.1t_amd64 package. deb

Can you please tell me how can I remove or replace the vulnerable package?

**Comment by DmitriyLewen** [comment]
Hello @partizanes 
Sorry for confusing you about vulnerabilities in 1.1.1t version.

I checked NVD and looks like `1.1.1t`(as separate package) doesn't contain these vulnerabilities.

But there is some problem to exclude these CVEs from result:

Ubuntu doesn't have information about `openssl 1.1.1x` package for `Jammy Ubuntu`.
This Ubuntu version uses `openssl 3.x.x`. This is why Ubuntu advisory database only include `3.x.x` fixed versions.
We can't get information about fixed version for `openssl 1.1.1x`.

This is special case, so i think you can [filter](https://aquasecurity.github.io/trivy/v0.41/docs/configuration/filtering/) these CVE or create [module](https://aquasecurity.github.io/trivy/v0.41/docs/advanced/modules/) to handle these CVE.

Regards, Dmitriy
